### PR TITLE
Feature/ph config

### PR DESCRIPTION
**Pull request by mcousillas6**

### CODE REVIEW

File: lib/bio_monitor/routine_monitor.ex [hunk 1]
```diff
@@ -158,6 +224,41 @@ defmodule BioMonitor.RoutineMonitor do
...
   end
 
+  defp is_offset_stable?(ph_cal) do
+    case Enum.count(ph_cal.values) >= 10 do
+      true ->
+        case Math.Enum.mean(ph_cal.values) do
+          nil -> false
+          mean ->
+            oscillation = mean - List.last(ph_cal.values)
+            oscillation <= @ph_oscillation_tolerance && oscillation >= -@ph_oscillation_tolerance
```
Comment (diegomedina248) [inline lib/bio_monitor/routine_monitor.ex]: `Kernel.abs(oscillation) <= @ph_oscillation_tolerance`?

File: lib/bio_monitor/sensor_manager.ex [hunk 2]
```diff
@@ -51,13 +51,23 @@ defmodule BioMonitor.SensorManager do
...
+      _ -> :error
+    end
+  end
+
   @doc """
     sets the offset of the ph sensor for calibration
   """
-  def set_ph_offset(offset) do
-    case send_command(:ph, "setPhOffset:#{offset}") do
+  def set_ph_offset(target, offser) do
```
Comment (diegomedina248) [inline lib/bio_monitor/sensor_manager.ex]: fix the typo or it wont work :)

File: lib/bio_monitor/sensor_manager.ex [hunk 3]
```diff
@@ -101,6 +111,28 @@ defmodule BioMonitor.SensorManager do
...
+      * {:error, message}
+  """
+  def send_and_read(sensor, command) do
+    with {:ok, gs_name} <- gs_name_for_sensor(sensor),
+      {:ok, result} <- SerialMonitor.send_and_read(gs_name, command)
+    do
+      {:ok, result}
+    else
+      {:error, message} ->
+        {:error, "Error sending command", message}
```
Comment (diegomedina248) [inline lib/bio_monitor/sensor_manager.ex]: Should be in Spanish right?

File: lib/bio_monitor/serial_monitor.ex [hunk 4]
```diff
@@ -183,6 +187,21 @@ defmodule BioMonitor.SerialMonitor do
...
+    case UART.write(state.serial_pid, command) do
+      :ok ->
+        case get_sensor_reading(state) |> parse_reading do
+          {:error, msg} ->
+            {:reply, {:error, msg}, state}
+          value ->
+            {:reply, {:ok, value}, state}
+        end
+      _ ->
+        {:reply, {:error, "Uknown error"}, state}
```
Comment (diegomedina248) [inline lib/bio_monitor/serial_monitor.ex]: same

File: web/controllers/ph_controller.ex [hunk 5]
```diff
@@ -14,24 +15,46 @@ defmodule BioMonitor.PhController do
...
+        conn |> render("cal_started.json")
+      :routine_in_progress ->
         conn
         |> put_status(:unprocessable_entity)
-        |> render(ErrorView, "error.json", %{message: message})
-      _ ->
+        |> render(
+          ErrorView,
+          "error.json",
+          %{message: "No se puede claibrar el sensor mientras este corriendo un experimento"}
```
Comment (diegomedina248) [inline web/controllers/ph_controller.ex]: typo: `calibrar`

File: lib/bio_monitor/routine_monitor.ex (hunk 1 above)
Comment (mcousillas6) [reply lib/bio_monitor/routine_monitor.ex]: I like it 👌

File: lib/bio_monitor/sensor_manager.ex (hunk 2 above)
Comment (mcousillas6) [reply lib/bio_monitor/sensor_manager.ex]: even worse, i was doing string interpolation with the javascript way (using $ instead of #) 😬

Comment (diegomedina248) [reply lib/bio_monitor/sensor_manager.ex]: oh haha i use that in kotlin, so it seemed fine

**Pull request merged**